activemq-client/src/main/java/org/apache/activemq/management/StringStatisticImpl.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.management;
+
+/**
+ * A string statistic implementation
+ */
+public class StringStatisticImpl extends UnsampledStatisticImpl implements StringStatistic {
+
+    // Note: [AMQ-8463] Adding volatile to 'value' would be more accurate, but performance impact
+    private String value = null;
```

Comment:
We should not be compromising correctness for performance. If this is multi-threaded you need to make it thread safe. Furthermore, the impact of volatile is going to be negligible in the real world.

You could use AtomicReference and make this final

Reply:
I'm thinking 'volatile' over AtomicReference. Same thread-safety, fewer objects created.

Reply:
I don't think the extra object creation matters in this case and the extra features of the atomic classes are nice. One thing i just realized was that if we want to use other Atomic classes like AtomicLong, which would be nice, then we may not be able to use generics after all as that atomic type needs to be different and we'd have slightly different code

Reply:
Depending on the use case this is what static atomic field updater types are for, all the features like CAS but only one static allocation.